tools/cli/app_test.go

```diff
@@ -558,9 +558,125 @@ func (s *cliAppSuite) TestObserveWorkflowWithID() {
 }
 
 func (s *cliAppSuite) TestParseTime() {
-	s.Equal(int64(100), parseTime("", 100))
-	s.Equal(int64(1528383845000000000), parseTime("2018-06-07T15:04:05+00:00", 0))
-	s.Equal(int64(1528383845000000000), parseTime("1528383845000000000", 0))
+	tests := []struct {
+		timeStr       string // input
+		defVal        int64  // input
+		lowerUnixNano int64  // lower range of expected output
+		upperUnixNano int64  // upper range of expected output
```

Comment:
How about removing `lowerUnixNano` and `upperUnixNano`,  but only had one `expected`, for all the new tests, you can assert `expected <= eval(text) <= expected + delta`. This will reduce some lines of code

Reply:
Yeah, even I thought of doing it, The only reason I did it, as the previously written tests had exact values, and I didn't want to change the passing criteria for those. So, I did it with a `lower` and `upper` bound, where the bounds coincide for previously written tests.

One other way I thought of was, I can have an `expected` value and a `delta` as part of the test cases, where I set `delta` value as `0` for previously written tests, and `5 ms` for new tests.

Or if we are not too strict with little change in passing criteria, we can just go ahead with what you suggested above. It surely improves the readability of the tests.

Let me know what you think is the best way. I will refactor the tests accordingly.

Reply:
I see you concerns. Let's have two tests, one for existing test cases using equal; one for new cases using `expected <= eval(text) <= expected + delta`

Reply:
Restored the original and added a second one for new changes, as you suggested 😃